fdbserver/Resolver.actor.cpp

```diff
@@ -482,7 +482,7 @@ ACTOR Future<Void> resolveBatch(Reference<Resolver> self,
 				reply.tpcvMap.clear();
 			} else {
 				std::set<uint16_t> writtenTLogs;
-				if (shardChanged || reply.privateMutationCount) {
+				if (shardChanged || toCommit->getEmptyMessageRatio() < 1.0) {
```

Comment:
Yes, this is causing the problem. `reply.privateMutationCount` was set to `toCommit->getMutationCount()`, which is:
```
uint32_t getMutationCount() const { return subsequence; }
```

The problem is in this function. Because `subsequence` is initialized to 1, so `reply.privateMutationCount` is always true. The proper fix should be `return subsequence -1` for this function.

Reply:
I think I got confused about the call to "getEmptyMessageRatio()" in your commit.